src/demo-app/tree/tree-demo.ts

```diff
@@ -0,0 +1,56 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import {Component} from '@angular/core';
+import {FlatTreeControl, NestedTreeControl} from '@angular/cdk/tree';
+import {of as ofObservable} from 'rxjs/observable/of';
+
+import {JsonNode, JsonDatabase} from './json-database';
+import {FlatDataSource, JsonFlatNode} from './flat-data-source';
+import {JsonNestedDataSource} from './nested-data-source';
+
+
+@Component({
+  moduleId: module.id,
+  selector: 'tree-demo',
+  templateUrl: 'tree-demo.html',
+  styleUrls: ['tree-demo.css'],
+})
+export class TreeDemo {
+  // Flat tree control
+  treeControl: FlatTreeControl<JsonFlatNode>;
+
+  // Nested tree control
+  nestedTreeControl: NestedTreeControl<JsonNode>;
+
+  // Flat tree data source
+  dataSource: FlatDataSource;
+
+  // Nested tree data source
+  nestedDataSource: JsonNestedDataSource;
+
+  constructor(database: JsonDatabase) {
+    // For flat tree
+    this.treeControl = new FlatTreeControl<JsonFlatNode>(this.getLevel, this.isExpandable);
```

Comment:
Would it make sense to use a more real-world example, such as collection of directories?